### PR TITLE
Drop defunct prevector compat handling

### PR DESCRIPTION
**Pull request by Empact**
This is clean-up post #14651:
* Use one implementation of `prevector::fill`, as it's possible now that the implementations are identical.
* Only apply the `IS_TRIVIALLY_CONSTRUCTIBLE` handling to the bench file where it is used, and drop the now-unnecessary associated compat includes.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

File: src/compat.h [hunk 1]
```diff
@@ -10,16 +10,6 @@
 #include <config/bitcoin-config.h>
 #endif
 
-#include <type_traits>
```
Comment (promag) [inline src/compat.h]: Should include in bench? https://en.cppreference.com/w/cpp/types/is_default_constructible

**Review by promag — COMMENTED** [review]
LGTM

**Comment by practicalswift** [comment]
Concept ACK

**Comment by laanwj** [comment]
concept ACK

File: src/bench/prevector.cpp [hunk 2]
```diff
@@ -2,13 +2,22 @@
...
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <compat.h>
 #include <prevector.h>
 #include <serialize.h>
 #include <streams.h>
 
 #include <bench/bench.h>
 
+#include <type_traits>
```
Comment (laanwj) [inline src/bench/prevector.cpp]: please group this with the other system includes at the top

**Comment by sipa** [comment]
utACK 69ca48717ceb31e37e90276278362c809cf98cc6

**Comment by MarcoFalke** [comment]
utACK 69ca48717ceb31e37e90276278362c809cf98cc6

**Comment by promag** [comment]
utACK 69ca487.

**Comment by practicalswift** [comment]
utACK 69ca48717ceb31e37e90276278362c809cf98cc6

**Comment by laanwj** [comment]
utACK 69ca48717ceb31e37e90276278362c809cf98cc6

**Pull request merged**